lib/platform/bitbucket/utils.js

```diff
@@ -0,0 +1,144 @@
+const url = require('url');
+const FormData = require('form-data');
+const api = require('./bb-got-wrapper');
+
+const repoInfoTransformer = repoInfoBody => ({
+  privateRepo: repoInfoBody.is_private,
+  isFork: !!repoInfoBody.parent,
+  repoFullName: repoInfoBody.full_name,
+  owner: repoInfoBody.owner.username,
+  mainbranch: repoInfoBody.mainbranch.name,
+  mergeMethod: 'merge',
+});
+
+const prStates = {
+  open: ['OPEN'],
+  notOpen: ['MERGED', 'DECLINED', 'SUPERSEDED'],
+  merged: ['MERGED'],
+  closed: ['DECLINED', 'SUPERSEDED'],
+  all: ['OPEN', 'MERGED', 'DECLINED', 'SUPERSEDED'],
+};
+
+const buildStates = {
+  success: 'SUCCESSFUL',
+  failed: 'FAILED',
+  pending: 'INPROGRESS',
+};
+
+const addMaxLength = (inputUrl, pagelen = 100) => {
+  const { search, ...parsedUrl } = url.parse(inputUrl, true);
+  const maxedUrl = url.format({
+    ...parsedUrl,
+    query: { ...parsedUrl.query, pagelen },
+  });
+  return maxedUrl;
+};
+
+const filesEndpoint = async (reqUrl, method = 'get', options) => {
+  const values = await accumulateValues(reqUrl, method, options);
+  const commitFolders = values.filter(
+    value => value.type === 'commit_directory'
+  );
+  let commitFiles = values.filter(value => value.type === 'commit_file');
+
+  if (commitFolders.length !== 0) {
+    const moreFiles = [].concat(
+      ...(await Promise.all(
+        commitFolders
+          .map(folder => folder.links.self.href)
+          .filter(Boolean)
+          .map(selfUrl => filesEndpoint(selfUrl, method, options))
+      ))
+    );
+    commitFiles = [...moreFiles, ...commitFiles];
+  }
+
+  return commitFiles;
+};
+
+const accumulateValues = async (reqUrl, method = 'get', options, pagelen) => {
+  let accumulator = [];
+  let nextUrl = addMaxLength(reqUrl, pagelen);
+  const lowerCaseMethod = method.toLocaleLowerCase();
+
+  while (typeof nextUrl !== 'undefined') {
+    const { body } = await api[lowerCaseMethod](nextUrl, options);
+    accumulator = [...accumulator, ...body.values];
+    nextUrl = body.next;
+  }
+
+  return accumulator;
+};
+
+const isConflicted = files => {
```

Comment:
if this functions doesnt use `parse-diff` anymore, then I would suggest to remove it from the deps

Reply:
Is it not necessary anymore, or removed by accident?

Reply:
I believe I used it for parsing conflicts in `isConflicted` function.

Reply:
as far as I can see its not used anymore

Reply:
It's used in `isPrConflicted(prNo)`